### PR TITLE
[BEAM-7326] add documentation bigquery data types

### PR DESCRIPTION
**Pull request by Juta**
Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by Juta** [comment]
R: @tvalentyn

File: website/src/documentation/io/built-in-google-bigquery.md [hunk 1]
```diff
@@ -163,6 +163,29 @@ table that you want to write to, unless you specify a [create
 disposition](#create-disposition) of `CREATE_NEVER`. [Creating a table
 schema](#creating-a-table-schema) covers schemas in more detail.
 
+### Data types
```
Comment (tvalentyn) [inline website/src/documentation/io/built-in-google-bigquery.md]: We can preview these changes by looking at staged Website (one of precommit checks). We need to fix formatting here, see: http://apache-beam-website-pull-requests.storage.googleapis.com/8873/documentation/io/built-in/google-bigquery/index.html#data-types.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java [hunk 2]
```diff
@@ -142,6 +143,34 @@
...
+ * {@link TableCell} instances.
+ *
+ * <p>{@link TableCell}: Holds the value for one cell (or field). Has one attribute, 'v', which is
+ * the value of the table cell.
+ *
+ * <p>As of Beam 2.7.0, the NUMERIC data type is supported. This data type supports high-precision
+ * decimal numbers (precision of 38 digits, scale of 9 digits). The GEOGRAPHY data type works with
+ * Well-Known Text (See <a href="https://en.wikipedia.org/wiki/Well-known_text">
+ * https://en.wikipedia.org/wiki/Well-known_text</a>) format for reading and writing to BigQuery.
+ * The BYTES data type requires that bytes are encoded using base64 encoding when writing to
```
Comment (tvalentyn) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java]: How about the following wording: 
```BigQuery IO requires values of BYTES datatype to be encoded using base64 encoding when writing to BigQuery. When bytes are read from BigQuery they are returned as base64-encoded strings.```

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 3]
```diff
@@ -224,6 +224,9 @@ def compute_table_name(row):
 The GEOGRAPHY data type works with Well-Known Text (See
 https://en.wikipedia.org/wiki/Well-known_text) format for reading and writing
 to BigQuery.
+The BYTES data type requires that bytes are encoded using base64 encoding when
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: How about the following wording: 
```BigQuery IO requires values of BYTES datatype to be encoded using base64 encoding when writing to BigQuery. When bytes are read from BigQuery they are returned as base64-encoded bytes.``` 
note: looks like in Java bytes are returned as strings, but in Python as bytes, so there is a difference with Java wording.

File: website/src/documentation/io/built-in-google-bigquery.md [hunk 4]
```diff
@@ -163,6 +163,29 @@ table that you want to write to, unless you specify a [create
...
+
+BigQuery supports the following data types: STRING, BYTES, INTEGER, FLOAT,
+NUMERIC, BOOLEAN, TIMESTAMP, DATE, TIME, DATETIME and GEOGRAPHY.
+All possible values are described at [https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types] (https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types).
+BigQueryIO allows you to use all of these data types. The following examples
+shows the correct format for data types used when reading from and writing to
```
Comment (tvalentyn) [inline website/src/documentation/io/built-in-google-bigquery.md]: s/shows/show

File: website/src/documentation/io/built-in-google-bigquery.md [hunk 5]
```diff
@@ -163,6 +163,29 @@ table that you want to write to, unless you specify a [create
...
+{% github_sample /apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryDataTypes
+%}```
+```py
+{% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets.py tag:model_bigqueryio_data_types
+%}```
+
+As of Beam 2.7.0, the NUMERIC data type is supported. This data type supports
+high-precision decimal numbers (precision of 38 digits, scale of 9 digits).
+The GEOGRAPHY data type works with Well-Known Text (See ([https://en.wikipedia.org/wiki/Well-known_text](https://en.wikipedia.org/wiki/Well-known_text))
+format for reading and writing to BigQuery.
```
Comment (tvalentyn) [inline website/src/documentation/io/built-in-google-bigquery.md]: We can update the Python wording here as well.

File: website/src/documentation/io/built-in-google-bigquery.md [hunk 6]
```diff
@@ -163,6 +163,29 @@ table that you want to write to, unless you specify a [create
...
+
+BigQuery supports the following data types: STRING, BYTES, INTEGER, FLOAT,
+NUMERIC, BOOLEAN, TIMESTAMP, DATE, TIME, DATETIME and GEOGRAPHY.
+All possible values are described at [https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types] (https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types).
+BigQueryIO allows you to use all of these data types. The following examples
+shows the correct format for data types used when reading from and writing to
+BigQuery:
+
+```java
+{% github_sample /apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryDataTypes
```
Comment (tvalentyn) [inline website/src/documentation/io/built-in-google-bigquery.md]: This code snippet shows empty in the preview link.

Comment (Juta) [reply website/src/documentation/io/built-in-google-bigquery.md]: I don't see a mistake in the syntax, is it possible that the preview actually looks at the master branch and this explains why the code snippet is empty?

**Comment by Juta** [comment]
Thanks @tvalentyn, PTAL

File: website/src/documentation/io/built-in-google-bigquery.md [hunk 7]
```diff
@@ -163,6 +163,43 @@ table that you want to write to, unless you specify a [create
...
+base64 encoded strings.
+
+<!-- Python specific -->
+
+{:.language-py}
+As of Beam 2.7.0, the NUMERIC data type is supported. This data type supports
+high-precision decimal numbers (precision of 38 digits, scale of 9 digits).
+The GEOGRAPHY data type works with Well-Known Text (See [https://en.wikipedia.org/wiki/Well-known_text](https://en.wikipedia.org/wiki/Well-known_text)
+format for reading and writing to BigQuery.
+The BYTES data type requires that bytes are encoded using base64 encoding when
```
Comment (tvalentyn) [inline website/src/documentation/io/built-in-google-bigquery.md]: Let's match the wording `BigQuery IO requires ...`. Also, please mark "done" the resolved comment threads. Thank you!

File: website/src/documentation/io/built-in-google-bigquery.md (hunk 4 above)
Comment (tvalentyn) [reply website/src/documentation/io/built-in-google-bigquery.md]: done

File: website/src/documentation/io/built-in-google-bigquery.md (hunk 6 above)
Comment (tvalentyn) [reply website/src/documentation/io/built-in-google-bigquery.md]: Hi @melap - does the code snippet change formatting look good to you? For some reason we don't see it in the preview.

**Comment by tvalentyn** [comment]
R: @chamikaramj

File: website/src/documentation/io/built-in-google-bigquery.md (hunk 1 above)
Comment (Juta) [reply website/src/documentation/io/built-in-google-bigquery.md]: I fixed the problem with the link not displaying correctly, not sure if there is an other issue?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java (hunk 2 above)
Comment (Juta) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java]: ok

File: sdks/python/apache_beam/io/gcp/bigquery.py (hunk 3 above)
Comment (Juta) [reply sdks/python/apache_beam/io/gcp/bigquery.py]: ok

File: website/src/documentation/io/built-in-google-bigquery.md (hunk 5 above)
Comment (Juta) [reply website/src/documentation/io/built-in-google-bigquery.md]: done

File: website/src/documentation/io/built-in-google-bigquery.md (hunk 7 above)
Comment (Juta) [reply website/src/documentation/io/built-in-google-bigquery.md]: done

Comment (Juta) [reply website/src/documentation/io/built-in-google-bigquery.md]: done

**Comment by chamikaramj** [comment]
Thanks. This is great. LGTM.

**Comment by tvalentyn** [comment]
Thanks, @Juta and @chamikaramj !

**Pull request merged**